client/lxd.go

```diff
@@ -62,6 +70,28 @@ func (r *ProtocolLXD) GetHTTPClient() (*http.Client, error) {
 	return r.http, nil
 }
 
+// Do performs a Request, using macaroon authentication if set.
+func (r *ProtocolLXD) Do(req *http.Request) (*http.Response, error) {
```

Comment:
Is there a reason for this to be a public function?

Reply:
not really, made it private